WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewModel.swift

```diff
@@ -75,6 +80,17 @@ final class SetUpTapToPayInformationViewModel: PaymentSettingsFlowPresentedViewM
         .store(in: &subscriptions)
     }
 
+    func setUpTapped() {
+        setUpInProgress = true
+        connectionController?.searchAndConnect { _ in
+            /// No need for logic here. Once connected, the connected reader will publish
+            /// through the `cardReaderAvailableSubscription`, so we can just
+            /// dismiss the connection flow alerts.
```

Comment:
Just checking if I understand the logic at this point: This closure can return a result such as `Result<CardReaderConnectionResult, Error>`, however we don't really need to return any specific result at this point, as we'll have a publisher that deals with state updates (for example dismissing) as needed. Is that correct?

Reply:
Yeah, that's right. The flow moves on to the next step (currently a blank screen, soon a "you're ready to go" celebration page) when there's a successful connection, and it does that by other classes observing the connected readers. Failures/retries or cancellations are dealt with by the connection controller internally.